src/collision.cpp

```diff
@@ -614,3 +489,212 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 
 	return result;
 }
+
+bool collisionCheckIntersection(Environment *env, IGameDef *gamedef,
+		const aabb3f &box_0, const v3f &pos_f, ActiveObject *self,
+		bool collideWithObjects)
+{
+	#define PROFILER_NAME(text) (s_env ? ("Server: " text) : ("Client: " text))
+	Map *map = &env->getMap();
+	ServerEnvironment *s_env = dynamic_cast<ServerEnvironment*>(env);
+
+	ScopeProfiler sp(g_profiler, PROFILER_NAME("collisionInCollision()"), SPT_AVG);
+
+	/*
+		Collect node boxes
+	*/
+	std::vector<NearbyCollisionInfo> cinfo;
+	{
+		//TimeTaker tt2("collisionMoveSimple collect boxes");
+		ScopeProfiler sp2(g_profiler, PROFILER_NAME("collision collect boxes"), SPT_AVG);
+
+		v3s16 min = floatToInt(pos_f + box_0.MinEdge, BS) - v3s16(1, 1, 1);
+		v3s16 max = floatToInt(pos_f + box_0.MaxEdge, BS) + v3s16(1, 1, 1);
+
+		bool any_position_valid = add_area_node_boxes(min, max, gamedef, map, cinfo);
+
+		// Do not move if world has not loaded yet, since custom node boxes
+		// are not available for collision detection.
+		// This also intentionally occurs in the case of the object being positioned
+		// solely on loaded CONTENT_IGNORE nodes, no matter where they come from.
+		if (!any_position_valid) {
+			return true;
+		}
+	} // tt2
+
+	if(collideWithObjects)
+	{
+		/* add object boxes to cinfo */
```

Comment:
unhelpful comment, which you've also duplicated three times

Reply:
Fixed also in `collisionMoveSimple`.